### ISSUE TITLE
low performance if fusion with inner and outer reductions segmented into two segments using combined scheduler and transpose scheduler due to limited shared memory and register

### ISSUE THREAD

**Issue by liqiangxl** [NVIDIA/Fuser#744]
When broadcast_in_dim is used in Python fusion definition, it is converted to broadcast and expand. When there is not enough shared memory and register for persistence, the fusion is segmented into two kernels: one is using the  combined scheduler and the other is using the transpose scheduler. This segmentation is not optimal due to **4 inter-segment tensors** and extra transpose. Ideally, it should be segmented into two segments using inner scheduler and outer scheduler, respestively.
```
Inputs:
  T457_g[ iS951{( ceilDiv(( ( ceilDiv(i0, 32) ) * ( ceilDiv(i1, 32) ) ), 1) )}, iS952{1}, iS1126{( ceilDiv(( ceilDiv(( 32 * 32 ), 8) ), 128) )}, iS1127{128}, iS1125{8} ], __half
  T459_g[ iS984{( ceilDiv(( ceilDiv(i0, 32) ), 1) )}, iS985{1}, iS1061{( ceilDiv(( ceilDiv(32, 8) ), 128) )}, iS1062{128}, iS1060{8} ], __half
  T461_g[ iS909{( ceilDiv(( ( ceilDiv(i3, 32) ) * ( ceilDiv(i4, 32) ) ), 1) )}, iS910{1}, iS1096{( ceilDiv(( ceilDiv(( 32 * 32 ), 8) ), 128) )}, iS1097{128}, iS1095{8} ], __half
  T463_g[ iS972{( ceilDiv(( ( ceilDiv(i8, 32) ) * ( ceilDiv(1, 32) ) ), 1) )}, iS973{1}, iS989{( ceilDiv(( ceilDiv(( 32 * 32 ), 8) ), 128) )}, iS990{128}, iS988{8} ], __half
  i0, nvfuser_index_t
Outputs:
  T35_g[ iblockIdx.x867{( ceilDiv(( ( ceilDiv(i8, 32) ) * ( ceilDiv(i1, 32) ) ), 1) )}, iUS868{1}, iUR1066{( ceilDiv(( ceilDiv(( 32 * 32 ), 8) ), 128) )}, ithreadIdx.x1067{128}, iV1065{8} ] ca_pos( 2 ) produce_pos( 2 ), __half

Inputs:
  T4_g[ iS1396{gridDim.y}, iS1397{( ceilDiv(i7, gridDim.y) )} ], float
  T1_g[ iS1168{gridDim.y}, iS1169{( ceilDiv(i3, gridDim.y) )}, iS1177{( ceilDiv(( ceilDiv(( ceilDiv(( ceilDiv(i4, 8) ), blockDim.x) ), 10) ), 1) )}, iS1173{blockDim.x}, iS1174{10}, iS1176{1}, iS1171{8} ], __half
  T0_g[ iS1038{gridDim.y}, iS1039{( ceilDiv(i0, gridDim.y) )}, iS1047{( ceilDiv(( ceilDiv(( ceilDiv(( ceilDiv(i1, 8) ), blockDim.x) ), 10) ), 1) )}, iS1043{blockDim.x}, iS1044{10}, iS1046{1}, iS1041{8} ], __half
  T2_g[ iS1425{( ceilDiv(( ceilDiv(( ceilDiv(( ceilDiv(i5, 8) ), blockDim.x) ), 10) ), 1) )}, iS1421{blockDim.x}, iS1422{10}, iS1424{1}, iS1419{8} ], __half
  T5_g[ iS1400{gridDim.y}, iS1401{( ceilDiv(i8, gridDim.y) )} ], float
  i7, nvfuser_index_t
  i8, nvfuser_index_t
  i5, nvfuser_index_t
  i0, nvfuser_index_t
Outputs:
  T37_g[ iblockIdx.y1462{gridDim.y}, ithreadIdx.x1460{blockDim.x}, iS1461{( ceilDiv(( ceilDiv(( ceilDiv(i1, 4) ), blockDim.x) ), gridDim.y) )}, iV1458{4} ] ca_pos( 3 ) produce_pos( 3 ), __half
  T36_g[ iblockIdx.y1482{gridDim.y}, ithreadIdx.x1480{blockDim.x}, iS1481{( ceilDiv(( ceilDiv(( ceilDiv(i1, 4) ), blockDim.x) ), gridDim.y) )}, iV1478{4} ] ca_pos( 3 ) produce_pos( 3 ), __half
  T457_g[ iblockIdx.y1364{gridDim.y}, iS1365{( ceilDiv(i0, gridDim.y) )}, ithreadIdx.y1373{( ceilDiv(( ceilDiv(( ceilDiv(( ceilDiv(i1, 8) ), blockDim.x) ), 10) ), 1) )}, ithreadIdx.x1369{blockDim.x}, iS1370{10}, iUS1372{1}, iV1367{8} ] ca_pos( 6 ) produce_pos( 6 ), __half
  T459_g[ iblockIdx.y1392{gridDim.y}, iS1393{( ceilDiv(i0, gridDim.y) )} ] ca_pos( 2 ) produce_pos( 2 ), __half
  T461_g[ iblockIdx.y1288{gridDim.y}, iS1289{( ceilDiv(i3, gridDim.y) )}, ithreadIdx.y1297{( ceilDiv(( ceilDiv(( ceilDiv(( ceilDiv(i4, 8) ), blockDim.x) ), 10) ), 1) )}, ithreadIdx.x1293{blockDim.x}, iS1294{10}, iUS1296{1}, iV1291{8} ] ca_pos( 6 ) produce_pos( 6 ), __half
  T463_g[ iblockIdx.y1108{gridDim.y}, iS1109{( ceilDiv(i8, gridDim.y) )}, bthreadIdx.y1117{( ceilDiv(( ceilDiv(( ceilDiv(( ceilDiv(( (nvfuser_index_t)(20480) ), 8) ), blockDim.x) ), 10) ), 1) )}, bthreadIdx.x1113{blockDim.x}, bS1114{10}, bUS1116{1}, bS1111{8} ] ca_pos( 2 ) produce_pos( 2 ), __half
```

repro:
```
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=DataType.Half, is_cpu=False)
    T1 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=DataType.Half, is_cpu=False)
    T2 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Float, is_cpu=False)
    T3 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Float, is_cpu=False)
    T4 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Half, is_cpu=False)
    T5 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Half, is_cpu=False)
    T6 = fd.ops.cast(T0, dtype=DataType.Float)
    T7 = fd.ops.cast(T1, dtype=DataType.Float)
    T8 = fd.ops.cast(T4, dtype=DataType.Float)
    T9 = fd.ops.cast(T5, dtype=DataType.Float)
    S10 = fd.define_scalar(20480, dtype=DataType.Int)
    T11 = fd.ops.broadcast_in_dim(T2, output_shape=[216, 20480], broadcast_dims=[0])
    T12 = fd.ops.sub(T7, T11)
    T13 = fd.ops.broadcast_in_dim(T3, output_shape=[216, 20480], broadcast_dims=[0])
    T14 = fd.ops.mul(T12, T13)
    T15 = fd.ops.broadcast_in_dim(T8, output_shape=[216, 20480], broadcast_dims=[1])
    T16 = fd.ops.mul(T6, T15)
    T17 = fd.ops.mul(S10, T16)
    T18 = fd.ops.sum(T16, axes=[1], keepdim=False, dtype=DataType.Null)
    T19 = fd.ops.broadcast_in_dim(T18, output_shape=[216, 20480], broadcast_dims=[0])
    T20 = fd.ops.mul(T16, T14)
    T21 = fd.ops.sum(T20, axes=[1], keepdim=False, dtype=DataType.Null)
    T22 = fd.ops.broadcast_in_dim(T21, output_shape=[216, 20480], broadcast_dims=[0])
    T23 = fd.ops.mul(T14, T22)
    T24 = fd.ops.sub(T17, T19)
    T25 = fd.ops.sub(T24, T23)
    S26 = fd.ops.reciprocal(S10)
    T27 = fd.ops.mul(S26, T13)
    T28 = fd.ops.mul(T27, T25)
    T29 = fd.ops.mul(T6, T14)
    T30 = fd.ops.sum(T29, axes=[0], keepdim=False, dtype=DataType.Null)
    T31 = fd.ops.sum(T6, axes=[0], keepdim=False, dtype=DataType.Null)
    T32 = fd.ops.cast(T28, dtype=DataType.Half)
    T33 = fd.ops.cast(T30, dtype=DataType.Half)
    T34 = fd.ops.cast(T31, dtype=DataType.Half)
    fd.add_output(T32)
    fd.add_output(T33)
    fd.add_output(T34)

with FusionDefinition() as fd:
    nvfuser_fusion_id0(fd)

inputs = [
    torch.randn((216, 20480), dtype=torch.float16, device='cuda:0').as_strided((216, 20480), (20480, 1)),
    torch.randn((216, 20480), dtype=torch.float16, device='cuda:0').as_strided((216, 20480), (20480, 1)),
    torch.randn((216,), dtype=torch.float32, device='cuda:0').as_strided((216,), (1,)),
    torch.randn((216,), dtype=torch.float32, device='cuda:0').as_strided((216,), (1,)),
    torch.randn((20480,), dtype=torch.float16, device='cuda:0').as_strided((20480,), (1,)),
    torch.randn((20480,), dtype=torch.float16, device='cuda:0').as_strided((20480,), (1,)),
]
fd.execute(inputs)
```
Performance if segmented into combined part and transpose part:
```
Launch Parameters: BlockDim.x = 64, BlockDim.y = 4, BlockDim.z = -1, GridDim.x = -1, GridDim.y = 108, GridDim.z = -1, Smem Size = 1024
blocks_per_sm= 1, warps_per_sm= 8, occupancy= 12.50%
kernel2 run in 0.07168 ms, achieved: 618.887 GB/s
Launch Parameters: BlockDim.x = 128, BlockDim.y = -1, BlockDim.z = -1, GridDim.x = 4480, GridDim.y = -1, GridDim.z = -1, Smem Size = 2304
blocks_per_sm= 12, warps_per_sm= 48, occupancy= 75.00%
kernel1 run in 0.026624 ms, achieved: 1329.25 GB/s
```
Performance if enforce segment into inner part and outer part:
```
Launch Parameters: BlockDim.x = 192, BlockDim.y = 2, BlockDim.z = -1, GridDim.x = 107, GridDim.y = 3, GridDim.z = -1, Smem Size = 1536
blocks_per_sm= 2, warps_per_sm= 24, occupancy= 37.50%
kernel2 run in 0.03072 ms, achieved: 1154.72 GB/s
Launch Parameters: BlockDim.x = 256, BlockDim.y = -1, BlockDim.z = -1, GridDim.x = 216, GridDim.y = -1, GridDim.z = -1, Smem Size = 1024
blocks_per_sm= 1, warps_per_sm= 8, occupancy= 12.50%
kernel1 run in 0.022528 ms, achieved: 1572.73 GB/s
```
